src/components/countries-borders-layer/countries-borders-layer.js

```diff
@@ -1,83 +1,86 @@
-import { useEffect } from 'react';
+import { useEffect, useState } from 'react';
+import { loadModules } from 'esri-loader';
 import { connect } from 'react-redux';
 import { debounce } from 'lodash';
+// CONSTANTS
+import { GRID_CELL_STYLES } from 'constants/graphic-styles';
+import { COUNTRIES_GENERALIZED_BORDERS_FEATURE_LAYER as bordersLayerTitle, GRAPHIC_LAYER } from 'constants/layers-slugs';
+// UTILS
+import { createGraphic, createGraphicLayer } from 'utils/graphic-layer-utils';
+import { hitResults, setCursor, drawGeometry, flyToGeometry, toggleCountryTooltip } from 'utils/globe-events-utils';
+// ACTIONS
 import * as urlActions from 'actions/url-actions';
-import { COUNTRIES_GENERALIZED_BORDERS_FEATURE_LAYER } from 'constants/layers-slugs';
 
-const actions = {...urlActions}
+const CountriesBordersLayerContainer = ({ view, changeGlobe, countryISO }) => {
 
-const CountriesBordersLayerContainer = ({ view, changeGlobe, countryTooltip, highlightedCountryIso }) => {
-  const getBordersLayer = (results) => {
-    if (!results.length) return null;
-    return results.find(result => result.graphic.layer.id === COUNTRIES_GENERALIZED_BORDERS_FEATURE_LAYER)
-  }
+
+
+  const [selectedCountryBorderGraphic, setSelectedCountryGraphic] = useState(null);
+  const [hoveredCountryBorderGraphic, setHoveredCountryGraphic] = useState(null);
+
+  //Create the graphics layer on mount
+  useEffect(() => {
+    loadModules(["esri/Graphic","esri/layers/GraphicsLayer"]).then(([Graphic, GraphicsLayer]) => {
+        const _selectedCountryBorderGraphic = createGraphic(Graphic, GRID_CELL_STYLES);
+        const _hoveredCountryBorderGraphic = createGraphic(Graphic, GRID_CELL_STYLES);
+        const graphicsLayer = createGraphicLayer(GraphicsLayer, [_selectedCountryBorderGraphic, _hoveredCountryBorderGraphic], GRAPHIC_LAYER);
+        setSelectedCountryGraphic(_selectedCountryBorderGraphic);
+        setHoveredCountryGraphic(_hoveredCountryBorderGraphic);
+        view.map.add(graphicsLayer);
+      })
+  }, [])
   
-  const onClickHandler = (bordersLayer) => {
-    if (bordersLayer) {
-      changeGlobe({countryTooltip: null});
-      const { graphic } = bordersLayer;
-      const { attributes } = graphic;
-      if (!countryTooltip || countryTooltip !== attributes.GID_0) {
-        view.goTo(graphic.geometry)
-        changeGlobe({countryTooltip: attributes.GID_0, countryName: attributes.NAME_0});
-      } 
-    } else if (countryTooltip) {
-      console.log('outside labels layer')
-      changeGlobe({countryTooltip: null});
-    }
+  const onClickHandler = bordersLayerFeatures => {
+    flyToGeometry(view, bordersLayerFeatures);
```

Comment:
Maybe we can wait for this to succeed before toggling the country tooltip. Not sure why it was not working fast enough on my side

Reply:
Yeah, it happens to me sometimes too. I'm not sure why it has this lag (it is an internal arcgis API function `view.goTo`). But yeah something to investigate in order to try to improve it